src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -108,12 +95,54 @@ public ODataBatchReaderState State
                 return this.batchReaderState;
             }
 
-            private set
+            protected set
             {
                 this.batchReaderState = value;
             }
         }
 
+        /// <summary>
+        /// >The batch-specific URL converter that stores the content IDs found in a changeset and supports resolving cross-referencing URLs.
+        /// </summary>
+        internal ODataBatchPayloadUriConverter PayloadUriConverter
+        {
+            get { return this.payloadUriConverter; }
+        }
+
+        /// <summary>
+        /// Reader's input context.
+        /// </summary>
+        protected ODataInputContext InputContext
+        {
+            get { return this.inputContext; }
+        }
+
+        /// <summary>
+        /// Previously cache contentId that should be applied to the next message read.
+        /// </summary>
+        protected string ContentIdToAddOnNextRead

```

Comment:
>ContentIdToAddOnNextRead [](start = 25, length = 24)

Can we just call this "CurrentContentId"?  The fact that it gets added to the URL Resolver is an implementation detail.

Reply:
We can, but I don't have strong reason to rename this since it helps the understanding of usage even when it is private in current master branch. See current implementation [https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/ODataBatchReader.cs#L56](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/ODataBatchReader.cs#L56).
Resolving it now. If you have strong opinion, feel free to re-open.

---
In reply to: [140932103](https://github.com/OData/odata.net/pull/960#discussion_r140932103) [](ancestors = 140932103)

Reply:
This used to be private, so the name didn't matter. Now that it's protected, it's exposed, so we should make sure the name is meaningful outside of the implementation.

---
In reply to: [142558638](https://github.com/OData/odata.net/pull/960#discussion_r142558638) [](ancestors = 142558638,140932103)